vaadin-combo-box-flow-parent/vaadin-combo-box-flow/src/main/java/com/vaadin/flow/component/combobox/ComboBox.java

```diff
@@ -1456,6 +1456,83 @@ public boolean isClearButtonVisible() {
         return super.isClearButtonVisibleBoolean();
     }
 
+    /**
+     * Supply items lazily with a callback from a backend. The component will
+     * automatically fetch more items and adjust its size until the backend runs
+     * out of items. Usage example without component provided filter:
+     * <p>
+     * {@code component.setItems(query ->
+     *             orderService.getOrders(query.getOffset(), query.getLimit());}
+     * <p>
+     * Since ComboBox supports filtering, it can be fetched via
+     * query.getFilter():
+     * <p>
+     * {@code component.setItems(query ->
+     *             orderService.getOrders(query.getFilter(), query.getOffset(),
+     *                                      query.getLimit());}
+     * <p>
+     * The returned data view object can be used for further configuration, or
+     * later on fetched with {@link #getLazyDataView()}. For using in-memory
+     * data, like {@link Collection}, use
+     * {@link HasListDataView#setItems(Collection)} instead.
+     * <p>
+     * Custom filter type converters can be used, see:
```

Comment:
Maybe add some more explanations: 
`If you want to filter the items in `ComboBox` by some value type other than a String, and your backend is able to fetch the items by this value, you can set the filter converter along with fetch callback.`